File: src/main/java/com/dongtronic/diabot/commands/DiacastCommand.java
package com.dongtronic.diabot.commands;

import com.dongtronic.diabot.exceptions.NoSuchEpisodeException;
import com.jagrosh.jdautilities.command.CommandEvent;
import com.jagrosh.jdautilities.doc.standard.CommandInfo;
import com.rometools.rome.feed.synd.SyndContent;
import com.rometools.rome.feed.synd.SyndEntry;
import com.rometools.rome.feed.synd.SyndFeed;
import com.rometools.rome.io.FeedException;
import com.rometools.rome.io.SyndFeedInput;
import com.rometools.rome.io.XmlReader;
import net.dv8tion.jda.core.EmbedBuilder;
import org.apache.commons.lang3.StringUtils;
import org.jdom2.Attribute;
import org.jdom2.Element;

import java.io.IOException;
import java.net.URL;
import java.util.List;

@CommandInfo(
    name = {"Diacast"},
    description = "Get information about a diacast episode"
)
public class DiacastCommand extends DiabotCommand {

  public DiacastCommand(Category category) {
    this.name = "diacast";
    this.help = "Get information about a diacast episode";
    this.guildOnly = true;
    this.category = category;
    this.examples = new String[]{"diabot diacast", "diabot diacast 6"};
  }

  private static String url = "https://diacast.xyz/?format=rss";

  @Override
  protected void execute(CommandEvent event) {
    try {
      String[] args = event.getArgs().split("\\s+");
      int episodeNumber = 0;

      if(args.length > 0 && StringUtils.isNumeric(args[0])) {
        episodeNumber = Integer.valueOf(args[0]);
      }

      SyndEntry episode = getEpisode(episodeNumber);

      EmbedBuilder builder = new EmbedBuilder();

      buildEpisodeCard(episode, builder);

      event.reply(builder.build());

    } catch (Exception ex) {
      event.replyError("Something went wrong: " + ex.getMessage());
    }
  }

  private void buildEpisodeCard(SyndEntry episode, EmbedBuilder builder) {
    builder.setTitle(episode.getTitle(), episode.getLink());
    builder.setAuthor("Diacast");

    for(Element element : episode.getForeignMarkup()) {
      if (element.getName().equals("summary")) {
        builder.setDescription(element.getValue());
      }

      if (element.getName().equals("image")) {
        String imageUrl = element.getAttributeValue("href");
        builder.setThumbnail(imageUrl);
      }

    }
  }

  private List<SyndEntry> getEpisodes() throws FeedException, IOException {
    URL feedSource = new URL(url);
    SyndFeedInput input = new SyndFeedInput();
    SyndFeed feed = input.build(new XmlReader(feedSource));
    return feed.getEntries();
  }

  private SyndEntry getEpisode(int episode) throws NoSuchEpisodeException, IOException, FeedException {
    List<SyndEntry> episodes = getEpisodes();
    if(episode == 0) {
      return episodes.get(0);
    }

    for(SyndEntry entry : episodes) {
      for(Element element : entry.getForeignMarkup()) {
        if(element.getName().equals("episode")) {
          String number = element.getValue();
          if(Integer.valueOf(number).equals(episode)) {
            return entry;
          }
        }
      }
    }

    throw new NoSuchEpisodeException();
  }


}

File: src/main/java/com/dongtronic/diabot/commands/fun/DiacastCommand.kt
package com.dongtronic.diabot.commands.`fun`

import com.dongtronic.diabot.commands.DiabotCommand
import com.dongtronic.diabot.exceptions.NoSuchEpisodeException
import com.jagrosh.jdautilities.command.Command
import com.jagrosh.jdautilities.command.CommandEvent
import com.rometools.rome.feed.synd.SyndEntry
import com.rometools.rome.io.FeedException
import com.rometools.rome.io.SyndFeedInput
import com.rometools.rome.io.XmlReader
import net.dv8tion.jda.core.EmbedBuilder
import org.apache.commons.lang3.StringUtils

import java.io.IOException
import java.net.URL

class DiacastCommand(category: Command.Category) : DiabotCommand(category, null) {

    private val episodes: List<SyndEntry>
        @Throws(FeedException::class, IOException::class)
        get() {
            val feedSource = URL("https://diacast.xyz/?format=rss")
            val input = SyndFeedInput()
            val feed = input.build(XmlReader(feedSource))
            return feed.entries
        }

    init {
        this.name = "diacast"
        this.help = "Get information about a diacast episode"
        this.guildOnly = true
        this.examples = arrayOf("diabot diacast", "diabot diacast 6")
    }

    override fun execute(event: CommandEvent) {
        var episodeNumber = 0
        try {
            val args = event.args.split("\\s+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

            if (args.isNotEmpty() && StringUtils.isNumeric(args[0])) {
                episodeNumber = Integer.valueOf(args[0])
            }

            val episode = getEpisode(episodeNumber)

            val builder = EmbedBuilder()

            buildEpisodeCard(episode, builder)

            event.reply(builder.build())

        } catch (ex: NoSuchEpisodeException) {
            event.replyError("Episode $episodeNumber does not exist")
        }
        catch (ex: Exception) {
            event.replyError("Something went wrong: " + ex.message)
        }

    }

    private fun buildEpisodeCard(episode: SyndEntry, builder: EmbedBuilder) {
        builder.setTitle(episode.title, episode.link)
        builder.setAuthor("Diacast")

        for (element in episode.foreignMarkup) {
            if (element.name == "summary") {
                builder.setDescription(element.value)
            }

            if (element.name == "image") {
                val imageUrl = element.getAttributeValue("href")
                builder.setThumbnail(imageUrl)
            }

        }
    }

    @Throws(NoSuchEpisodeException::class, IOException::class, FeedException::class)
    private fun getEpisode(episode: Int): SyndEntry {
        val episodes = episodes

        if (episode == 0) {
            return episodes[0]
        }


        for (entry in episodes) {
            for (element in entry.foreignMarkup) {
                if (element.name == "episode") {
                    val number = element.value
                    if (Integer.valueOf(number) == episode) {
                        return entry
                    }
                }
            }
        }

        throw NoSuchEpisodeException()
    }


}
